src/adv.py

```diff
@@ -37,15 +38,75 @@
 # Main
 #
 
+messages = {
+    'welcome': "\nWelcome to THE ADVENTURE GAME\n",
+    'name_input': "What is your name, young grasshopper?\n",
+    'dir_input': "\nWhich direction do you want to move?\nN, E, S, W: ",
+    'warning_dir': "No such direction from this room."
+}
+
+print(messages["welcome"])
+
+name = input(messages["name_input"])
+
+location = room['outside']
+
+status = True
+
+
 # Make a new player object that is currently in the 'outside' room.
 
+player = Player(name, location)
+
 # Write a loop that:
 #
-# * Prints the current room name
-# * Prints the current description (the textwrap module might be useful here).
+# + Prints the current room name
+# + Prints the current description (the textwrap module might be useful here).
 # * Waits for user input and decides what to do.
 #
 # If the user enters a cardinal direction, attempt to move to the room there.
 # Print an error message if the movement isn't allowed.
 #
 # If the user enters "q", quit the game.
+
+while status:
+    print(f"\n{player}")
+
+    direction = input(messages["dir_input"]).upper()
+
+    if len(direction) >= 1:
+        if direction[0] == "N":
+            if hasattr(player.location, "n_to"):
```

Comment:
good use of the method hasattr. you didn't go down the laid of using getattr by have self.n_to and others in the player class